lib/registration.pm

```diff
@@ -119,12 +135,20 @@ sub fill_in_registration_data {
                     save_screenshot;
                 }
             }
-            send_key 'alt-n';    # next
-            while (check_screen('import-untrusted-gpg-key', 20)) {
-                record_soft_failure 'untrusted gpg key';
-                send_key 'alt-t', 2;
+            send_key 'alt-n';
+            # start addons/modules registration, it needs longer time if select multiple or all addons/modules
+            while (check_screen(['import-untrusted-gpg-key', 'yast_scc-pkgtoinstall', 'inst-addon'], 120)) {
+                if (match_has_tag('import-untrusted-gpg-key')) {
+                    record_soft_failure 'untrusted gpg key';
+                    send_key 'alt-t';
+                    next;
+                }
+                elsif (match_has_tag('inst-addon') || match_has_tag('yast_scc-pkgtoinstall')) {
+                    # it would show Add On Product screen if scc registration correctly during installation
+                    # it would show software install dialog if scc registration correctly by yast2 scc
+                    last;
```

Comment:
@coolo 120 seconds is the max timeout, and it marks addons/modules registration done if it matches the tag inst-addon or yast_scc-pkgtoinstall, so normally it would finish less than 120s and depend on how many addons/modules selected. also the timeout probabaly could be less due to the faster connection speed than Beijing.

@okurz i didn't miss your comment. during addons/modules registration, it will contacting server for each addon/module, so I did try the something before like this:

while (check_screen(['import-untrusted-gpg-key', 'contacting-registration-server'])) {
    if (match_has_tag('import-untrusted-gpg-key')) {
        record_soft_failure 'untrusted gpg key';
        send_key 'alt-t';
        next;
    }
    elsif (match_has_tag('contacting-registration-server')) {
        sleep 5;
        next;
    }
} 
the result: http://147.2.207.208/tests/1460
the problem is that it recorded many contacting-registration-server screens may due to my slow connection speed.